Makefile

```diff
@@ -56,7 +56,8 @@ prefix_install: installables
 	mkdir -p "$(PREFIX)/Frameworks" "$(PREFIX)/bin"
 	rsync -a --delete "$(TEMPORARY_FOLDER)$(FRAMEWORKS_FOLDER)/$(OUTPUT_FRAMEWORK)" "$(PREFIX)/Frameworks/"
 	cp -f "$(TEMPORARY_FOLDER)$(BINARIES_FOLDER)/carthage" "$(PREFIX)/bin/"
-	install_name_tool -rpath "/Library/Frameworks" "@executable_path/../Frameworks/$(OUTPUT_FRAMEWORK)/Versions/Current/Frameworks/"  "$(PREFIX)/bin/carthage"
+	install_name_tool -rpath "/Library/Frameworks" "@executable_path/../Frameworks" "$(PREFIX)/bin/carthage"
```

Comment:
`@executable_path/../Frameworks` is already set by `Mac-Base.xcconfig` and is duplicated after installation.
```terminal.sh-session
➜  8:40:46 git:(improve-makefile-rpath-handling) make prefix_install PREFIX=./prefix
…
➜  8:48:52 git:(improve-makefile-rpath-handling) ✗ otool -l prefix/bin/carthage
prefix/bin/carthage:
Mach header
      magic cputype cpusubtype  caps    filetype ncmds sizeofcmds      flags
 0xfeedfacf 16777223          3  0x80           2    44       5040 0x00200085
Load command 0
…
Load command 36
          cmd LC_RPATH
      cmdsize 32
         path @executable_path/. (offset 12)
Load command 37
          cmd LC_RPATH
      cmdsize 80
         path @executable_path/CarthageKit.framework/Versions/Current/Frameworks (offset 12)
Load command 38
          cmd LC_RPATH
      cmdsize 48
         path @executable_path/../Frameworks (offset 12)
Load command 39
          cmd LC_RPATH
      cmdsize 96
         path @executable_path/../Frameworks/CarthageKit.framework/Versions/Current/Frameworks (offset 12)
Load command 40
          cmd LC_RPATH
      cmdsize 48
         path @executable_path/../Frameworks (offset 12)
Load command 41
          cmd LC_RPATH
      cmdsize 40
         path @loader_path/../Frameworks (offset 12)
Load command 42
…
```
It would be better to remove `$ (inherited)` from `LD_RUNPATH_SEARCH_PATHS` of the `carthage` target, or use` -delete_rpath` instead of `-rpath` here.